src/hash_generator/hashAlgorithmUtils.js

```diff
@@ -0,0 +1,209 @@
+ */
+
+import { getTheTimeZoneOffsetFromDateString } from '../utils/utils';
+import { dlNormalizer } from './dlNormalizer';
+
+/**
+ * The list of fields that will be ignored in the pre-hashed string
+ * @type {string[]}
+ */
+export const toBeIgnored = ['recordTime', 'eventID'];
+
+/**
+ * Check if the parameter is a string
+ * @param obj
+ * @return true if the parameter is a string, false otherwise
+ */
+export const isAString = (obj) => typeof obj === 'string' || obj instanceof String;
+
+/**
+ * If the parameter is a string, it returns the string without the useless spaces at the beginning
+ * or at the end
+ * If the parameter isn't a string, it returns the parameter
+ *
+ * It useful for the rule n°5 of the algorithm
+ *
+ * @param {*} obj
+ * @return {*} - the obj if it isn't a string, the parameter without useless spaces if it is a
+ * string
+ */
+export const removeWhiteSpaceAtTheBeginningOrEndOfString = (obj) => {
+  if (!isAString(obj)) return obj;
+
+  return obj.replace(/^\s+|\s+$/g, '');
+};
+
+/**
+ * If present, any URN-based standard vocabulary value (starting with ‘urn:epcglobal:cbv’) SHALL be
+ * expressed in its corresponding GS1 Web Vocabulary URI equivalent
+ * (starting with ‘https://ns.gs1.org’). Example: ‘urn:epcglobal:cbv:bizstep:receiving’ -->
+ * ‘https://ns.gs1.org/cbv/BizStep-receiving’
+ *
+ * @param obj
+ * @returns {*} the obj if it isn't a string, the parameter updated to follow the rules if it is
+ * a string
+ */
+export const convertURNBasedVocabularyToURI = (obj) => {
+  if (!isAString(obj)) return obj;
+
+  return obj.replace('urn:epcglobal:cbv:bizstep:', 'https://ns.gs1.org/cbv/BizStep-')
+    .replace('urn:epcglobal:cbv:disp:', 'https://ns.gs1.org/cbv/Disp-')
+    .replace('urn:epcglobal:cbv:btt:', 'https://ns.gs1.org/cbv/BTT-')
+    .replace('urn:epcglobal:cbv:sdt:', 'https://ns.gs1.org/cbv/SDT-')
+    .replace('urn:epcglobal:cbv:er:', 'https://ns.gs1.org/cbv/ER-');
+};
+
+/**
+ * if present, EPC URIs (starting with ‘urn:epc:id’), EPC Class URIs (starting with ‘urn:epc:class’)
+ * or EPC Pattern URIs (starting with ‘urn:epc:idpat’) SHALL be converted into the corresponding
+ * canonical GS1 Digital Link URI (starting with ‘https://id.gs1.org’). Canonical GS1 Digital Link
+ * URIs are specified in [GS1 Digital Link: URI Syntax, release 1.2], section 4.11.
+ *
+ * @param obj
+ * @param {boolean} throwError - if set to true, it will throw an error if the event misses some
+ * fields for example. Otherwise, it won't throw an error and it will still return the generated id
+ * @returns {*} the obj if it isn't a string, the parameter updated to follow the rules if it is
+ * a string
+ */
+export const convertEpcUriToDlUri = (obj, throwError) => {
+  if (!isAString(obj)) return obj;
+
+  return dlNormalizer(obj, throwError);
+};
+
+/**
+ * Check if the parameter is a date
+ * @param obj
+ * @return true if the parameter is a date, false otherwise
+ */
+// eslint-disable-next-line  no-restricted-globals
```

Comment:
Why is this `eslint-disable-next-line` here?

Reply:
because `isNan` is working here but `Number.isNan` isn't